### ISSUE TITLE
Check `.removeAttribute()` in  `prefer-dom-node-dataset` rule

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1600]
[`prefer-dom-node-dataset`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-dom-node-dataset.md) only check `.setAttribute()` now.

`delete element.dataset.foo` is recommanded https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#setting_values

## Fail

```js
element.removeAttribute('data-foo');
```

## Pass

```js
delete element.dataset.foo;
```

**Comment by sindresorhus** [comment]
This is now accepted.
